data-entry/dataEntry.app.js

```diff
@@ -74,23 +74,80 @@
-                        if (key.simple) {
-                            (function(key) {
-                                key.getDomainValues().then(function success(values) {
-                                    dataEntryModel.domainValues[key.colset.columns[0].name] = [];
-                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name];
-
-                                    angular.forEach(values.data, function(value) {
-                                        var field = Object.keys(value)[0];
-                                        domainValues.push(value[field]);
-                                    });
-                                }, function error(response) {
-                                    console.log(response);
-                                });
-                            })(key);
+                    angular.forEach(foreignKeys, function(fkey) {
+                        // simple implies one column
+                        if (fkey.simple) {
+                            var ftable = fkey.key.table;
+                            var keyColumn = fkey.key.colset.columns[0];
+
+                            /* FIRST USE CASE: covered by default; display = key column */
+
+                            var pattern = "{" + keyColumn.name + "}";
+                            var displayColumns = [keyColumn];
+
+                            /* SECOND USE CASE: conditional if the table is tagged as a vocabulary */
+
+                            try {
+                                try {
+                                    var vocabAnnotation = ftable.annotations.get("tag:misd.isi.edu,2015:vocabulary");
+                                } catch (error) {
+                                    // no vocab annotation, do nothing
+                                }
+
+                                try {
+                                    var displayAnnotation = ftable.annotations.get("tag:misd.isi.edu,2015:display");
+                                } catch (error) {
+                                    // no display annotation, do nothing
+                                }
+
+                                if (vocabAnnotation) {
+                                    if (vocabAnnotation.content.term) {
+                                        var termColumn = ftable.columns.get(vocabAnnotation.content.term);
+                                        displayColumns.push(termColumn); // the array is now [keyColumn, termColumn]
+                                    }
+                                    // vocabulary term is undefined
+                                    else {
+                                        var ftableColumns = ftable.columns.all();
+                                        for (var i = 0, length = ftableColumns.length; i < length; i++) {
+                                            var uppColumnName = $filter('uppercase')(column.name);
```

Comment:
Let's use plain JavaScript whenever it is available, so use `...column.name.toUpperCase()` instead of that `$filter`.